docroot/modules/custom/prisoner_hub_prison_context/tests/src/ExistingSite/PrisonContextTest.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Drupal\Tests\prisoner_hub_prison_context\ExistingSite;
+
+use Drupal\taxonomy\Entity\Vocabulary;
+use weitzman\DrupalTestTraits\Entity\TaxonomyCreationTrait;
+use weitzman\DrupalTestTraits\ExistingSiteBase;
+
+/**
+ * Functional tests to ensure that prison context jsonapi paths work correctly.
+ *
+ * @group prisoner_hub_prison_context
+ */
+class PrisonContextTest extends ExistingSiteBase {
+
+  use TaxonomyCreationTrait;
+
+  /**
+   * A user with permission to administer site configuration.
+   *
+   * @var \Drupal\taxonomy\Entity\Term
+   */
+  protected $term;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $vocab = Vocabulary::load('prisons');
+
+    $this->term = $this->createTerm($vocab);
+    /* @var \Drupal\Core\Routing\RouteBuilderInterface $route_builder */
+    $route_builder = $this->container->get('router.builder');
+    $route_builder->rebuildIfNeeded();
+  }
+
+  /**
+   * Tests that prison context jsonapi routes have been setup.
+   */
+  public function testRoutes() {
+    /* @var \Drupal\Core\Routing\RouteProviderInterface $route_provider */
+    $route_provider = $this->container->get('router.route_provider');
+    $this->assertGreaterThan(0, $route_provider->getRoutesByPattern('/jsonapi/prison/{prison}/node/article')->count(), 'There is a least one prison context jsonapi route.');
```

Comment:
In the ADR we used the format `prisons/{prison}`, but here it's singular. `prisons/` feels like it fits better with the JSON:API naming scheme. Is this way easier/neater?

Reply:
Hmmm, another good point.
I guess I was sticking to how Drupal names other resources in JSON:API.  I.e. it's `/node/article` not `/nodes/articles`.
But should it be different here?  As we are talking about a relationship?

Reply:
:+1: makes sense. I'm ambivalent (as long as the ADR matches the implementation 😉)

Reply:
Have just updated the ADR